### PR TITLE
Deprecate `skip_exit_code` in `DockerOperator` and `KubernetesPodOperator`

### PR DESCRIPTION
**Pull request by eladkal**
To be consistent with https://github.com/apache/airflow/pull/30692

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/docker/operators/docker.py [hunk 1]
```diff
@@ -276,7 +277,10 @@ def __init__(
         self.log_opts_max_size = log_opts_max_size
         self.log_opts_max_file = log_opts_max_file
         self.ipc_mode = ipc_mode
-        self.skip_exit_code = skip_exit_code
+        if skip_exit_code:
```
Comment (uranusjr) [inline airflow/providers/docker/operators/docker.py]: Is `skip_exit_code=0` a valid use case? If it is, we’ll need to use `is not None` here.

Comment (eladkal) [reply airflow/providers/docker/operators/docker.py]: I'll add it just in case

File: airflow/providers/cncf/kubernetes/operators/pod.py [hunk 2]
```diff
@@ -361,7 +362,12 @@ def __init__(
         self.termination_grace_period = termination_grace_period
         self.pod_request_obj: k8s.V1Pod | None = None
         self.pod: k8s.V1Pod | None = None
-        self.skip_exit_code = skip_exit_code
+        if skip_exit_code is not None:
+            warnings.warn(
+                "skip_exit_code is deprecated. Please use skip_on_exit_code", DeprecationWarning, stacklevel=2
+            )
+            self.skip_on_exit_code = skip_exit_code
+        self.skip_on_exit_code = skip_on_exit_code
```
Comment (hussein-awala) [inline airflow/providers/cncf/kubernetes/operators/pod.py]: Same comment here:
`self.skip_on_exit_code = skip_on_exit_code` overrides `self.skip_on_exit_code = skip_exit_code` and replaces it by `None`, you need to put it in `else`

**Comment by potiuk** [comment]
My though exactly that we should do that :)

**Comment by potiuk** [comment]
Another follow up here: https://github.com/apache/airflow/pull/30738 - we missed ExternalPythonOperator when adding the 'skip_on_exit_code`

**Comment by potiuk** [comment]
BTW. I **gues** we could still add those `skip_on_exit_code` changes to 2.6.0 ? @ephraimbuddy @jedcunningham - WDYT? Those changes seems pretty safe and possible to cherry-pick

**Comment by eladkal** [comment]
@potiuk this one is providers only. I split the PRs one for core and one for providers to avoid issues with milestones

**Comment by potiuk** [comment]
> @potiuk this one is providers only. I split the PRs one for core and one for providers to avoid issues with milestones

Ah True. silly me :)

**Review by potiuk — APPROVED** [review]
NIT: Should we also add container handling ? Or you want to separate it?

**Pull request merged**